### PR TITLE
Improved UART TX test

### PR DESCRIPTION
**Pull request by CTetford**
Add configurable option to UART TX test to specify approximate degrees of test rotation

### CODE REVIEW

**Review by ClutchplateDude — DISMISSED** [review]

**Comment by andre-stefanov** [comment]
mh it is possible to enable both test types at the moment. we should use an enumeration for the test type (e.g. values defined in Constants.hpp like for other hardware stuff).

**Comment by andre-stefanov** [comment]
5° on DEC is good visible but on RA i had to really pay close attention when not seeing the oat from the front (which is the case for some builds).

**Review by andre-stefanov — CHANGES_REQUESTED** [review]
mh it is possible to enable both test types at the moment. we should use an enumeration for the test type (e.g. values defined in Constants.hpp like for other hardware stuff).

File: src/Mount.cpp [hunk 1]
```diff
@@ -3691,31 +3691,46 @@ DayTime Mount::calculateHa()
...
+    _driverRA->microsteps(RA_TRACKING_MICROSTEPPING == 1 ? 0 : RA_TRACKING_MICROSTEPPING);
 }
     #endif
 
     #if DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART
 void Mount::testDEC_UART_TX()
 {
-    int _speed    = 1000;  //microsteps per driver clock tick
-    int _duration = 1000;  //Duration to run in ms
+    int _speed = (DEC_STEPPER_SPEED / 2)
```
Comment (julianneswinoga) [inline src/Mount.cpp]: Local variables shouldn't start with an underscore

Comment (julianneswinoga) [inline src/Mount.cpp]: You should add `const` to these local variables

File: src/Mount.cpp [hunk 2]
```diff
@@ -3691,31 +3691,46 @@ DayTime Mount::calculateHa()
...
+                 / 0.715255737f;  //microsteps per driver clock tick (From TMC2209 datasheet: v[Hz] (microstep/s) = VACTUAL[2209] * 0.715Hz)
```
Comment (julianneswinoga) [inline src/Mount.cpp]: Could you move these comments to the line above the variable? That way you don't need to add line breaks in the middle of the expression

File: src/Mount.cpp [hunk 3]
```diff
@@ -3691,31 +3691,46 @@ DayTime Mount::calculateHa()
...
     testUART_vactual(_driverDEC, _speed, _duration);
+    LOGV2(DEBUG_STEPPERS, F("[STEPPERS]: uartTest: Switching DEC driver to microsteps(%d) after UART test"), DEC_GUIDE_MICROSTEPPING);
+    _driverDEC->microsteps(DEC_GUIDE_MICROSTEPPING == 1 ? 0 : DEC_GUIDE_MICROSTEPPING);
 }
     #endif
 
     #if RA_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART || DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART
 void Mount::testUART_vactual(TMC2209Stepper *driver, int _speed, int _duration)
-{
+{  //Runs motor at specified speed for specified duration in each dirction, allowing 0.5s to stop motion after each move
```
Comment (julianneswinoga) [inline src/Mount.cpp]: This comment should go above the function, ideally in docblock format:
```C++
/**
 * Runs motor at specified speed for specified duration in each direction, allowing 0.5s to stop motion after each move
 * @param[in,out] driver The stepper driver instance
 * @param[in] speed The speed to run the stepper motor at (in deg/s? idk)
 * @param[in] duration The amount of time to turn the stepper, in seconds
 */
void Mount::testUART_vactual(TMC2209Stepper *driver, int speed, int duration)
{
```

**Review by julianneswinoga — CHANGES_REQUESTED** [review]

**Review by andre-stefanov — APPROVED** [review]

**Review by andre-stefanov — DISMISSED** [review]
Oops

File: src/Mount.cpp (hunk 3 above)
Comment (CTetford) [reply src/Mount.cpp]: Done

File: src/Mount.cpp (hunk 2 above)
Comment (CTetford) [reply src/Mount.cpp]: Done

File: src/Mount.cpp (hunk 1 above)
Comment (CTetford) [reply src/Mount.cpp]: Done

File: src/Mount.cpp (hunk 3 above)
Comment (julianneswinoga) [reply src/Mount.cpp]: ? I don't see it changed haha

Comment (CTetford) [reply src/Mount.cpp]: Sorry - my bad, I skipped one lol

File: src/Mount.cpp [hunk 4]
```diff
@@ -3691,31 +3691,52 @@ DayTime Mount::calculateHa()
...
+    LOGV2(DEBUG_STEPPERS, F("[STEPPERS]: uartTest: Switching DEC driver to microsteps(%d) after UART test"), DEC_GUIDE_MICROSTEPPING);
+    _driverDEC->microsteps(DEC_GUIDE_MICROSTEPPING == 1 ? 0 : DEC_GUIDE_MICROSTEPPING);
 }
     #endif
 
     #if RA_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART || DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART
+/**
+ * Runs motor at specified speed for specified duration in each direction, allowing 0.5s to stop motion after each move
+ * @param[in,out] driver The stepper driver instance
+ * @param[in] speed The speed to run the stepper motor at (in deg/s? idk)
```
Comment (julianneswinoga) [inline src/Mount.cpp]: Could you modify the comment to reflect what the function does? I just wrote out a basic structure that you could fill in

**Review by julianneswinoga — CHANGES_REQUESTED** [review]

**Review by julianneswinoga — APPROVED** [review]

**Pull request merged**